indico/queries/teach_tasks.py

```diff
@@ -0,0 +1,244 @@
+        }
+    """
+
+    def __init__(
+        self, name: str, dataset_id: int, source_column_id: int, targets: List[str]
+    ):
+        questions = [
+            {"type": "ANNOTATION", "targets": targets, "keywords": [], "text": name,}
+        ]
+        super().__init__(
+            query=self.query,
+            variables={
+                "name": name,
+                "dataset_id": dataset_id,
+                "questions": questions,
+                "source_col_id": source_column_id,
+            },
+        )
+
+    def process_response(self, response):
+        try:
+            questionnaire = Questionnaire(
+                **super().process_response(response)["createQuestionnaire"]
+            )
+        except IndexError:
+            raise IndicoNotFound("Failed to create Questionnaire")
+        return questionnaire
+
+
+class CreateTeachTask(RequestChain):
+    """
+    Creates a labeled questionaire (teach task) for a dataset.
+
+    Args:
+        name (str): The name of the questionnaire.
+        csv_path (str): Path to a csv with columns containing text and json encoded labels.
+        dataset (Dataset): Dataset to create the questionnaire from.
+        num_examples (int): number of rows in the dataset in total.
+        text_column (str): Optional. The column name in the csv containing the text.
+        label_column (str): Optional. The column name in the csv containing the json encoded labels. 
+        
+    Returns:
+        Dataset object   
+    """
+
+    previous = None
+
+    def __init__(
+        self,
+        name: str,
+        csv_path: str,
+        dataset: Dataset,
+        num_examples: int,
+        text_column="text",
+        label_column="labels",
+    ):
+        self.dataset_id = dataset.id
+        self.dataset = dataset
+        csv = pd.read_csv(csv_path)
+        self.data = {
+            fn: json.loads(label) for fn, label in zip(csv["text"], csv["labels"])
+        }
+        self.targets = list(
+            set(t["label"] for sample in self.data.values() for t in sample)
+        )
+        self.name = name
+        self.num_examples = num_examples
+        super().__init__()
+
+    def requests(self):
+        yield CreateQuestionaire(
+            name=self.name,
+            dataset_id=self.dataset_id,
+            source_column_id=self.dataset.datacolumns[0].id,
+            targets=self.targets,
+        )
+        questionaire_id = self.previous.id
+        yield GetDataset(id=self.dataset_id)
+        labelset_id = self.previous.labelsets[0].id
+        yield GetQuestionaireExamples(
+            questionaire_id=questionaire_id, num_examples=self.num_examples
+        )
+        yield AddLabels(
+            target=[self.data.get(f.source) for f in self.previous],
+            dataset_id=self.dataset_id,
+            row_index=[d.row_index for d in self.previous],
+            labelset_id=labelset_id,
+        )
+
+        yield GetDataset(id=self.dataset_id)
```

Comment:
Typically the final return value of a request that creates a resource is the id or object of the resource that was created